spack/package.py

```diff
@@ -155,19 +162,22 @@ def url_for_version(self, version):
     conflicts("+netcdf3", when="+netcdf")
     conflicts("+openmp", when="%apple-clang")
 
-    depends_on("wget", type=("build"), when="@:3.1 +netcdf4")
-    depends_on("wget", type=("build"), when="@:3.1 +netcdf")
     depends_on("ip@5.1:", when="@develop +ipolates")
     depends_on("libaec@1.0.6:", when="@3.2: +aec")
     depends_on("netcdf-c", when="@3.2: +netcdf4")
-    depends_on("netcdf-c", when="@3.2: +netcdf")
     depends_on("jasper@:2", when="@3.2: +jasper")
-    depends_on("zlib-api", when="+png")
-    depends_on("libpng", when="+png")
-    depends_on("openjpeg", when="+openjpeg")
+    depends_on("zlib-api", when="@3.2: +png")
+    depends_on("libpng", when="@3.2: +png")
+    depends_on("openjpeg", when="@3.2: +openjpeg")
 
     # Use Spack compiler wrapper flags
     def inject_flags(self, name, flags):
+
+        @when("@:2 ^gmake@:4.1")
+
+        def patch(self):
```

Comment:
Sorry, to clarify, I meant this should come after the whole inject_flags() function. That said, it was okay where it was previously, so it should be fine if you revert that last commit.

Reply:
Fixed. I was getting an error due to the '\' character in the config.h while trying to build with spack and I was having trouble debugging it.

Reply:
I got the version range for gmake backwards, that should be `@4.2:` because it's only a problem with the newer versions.